test/Nerdbank.MessagePack.Analyzers.Tests/MigrationAnalyzerTests.cs

```diff
@@ -0,0 +1,196 @@
+			public class MyType
+			{
+				public string Name { get; set; }
+			}
+
+			public class {|NBMsgPack100:MyTypeFormatter|} : IMessagePackFormatter<MyType>
+			{
+				public MyType Deserialize(ref MessagePackReader reader, MessagePackSerializerOptions options)
+				{
+					if (reader.TryReadNil())
+					{
+						return null;
+					}
+
+					string name = null;
+					options.Security.DepthStep(ref reader);
+					try
+					{
+						int count = reader.ReadArrayHeader();
+						for (int i = 0; i < count; i++)
+						{
+							switch (i)
+							{
+								case 0:
+									name = options.Resolver.GetFormatterWithVerify<string>().Deserialize(ref reader, options);
+									break;
+								default:
+									reader.Skip();
+									break;
+								}
+						}
+
+						return new MyType { Name = name };
+					}
+					finally
+					{
+						reader.Depth--;
+					}
+				}
+
+				public void Serialize(ref MessagePackWriter writer, MyType value, MessagePackSerializerOptions options)
+				{
+					if (value is null)
+					{
+						writer.WriteNil();
+						return;
+					}
+
+					writer.WriteArrayHeader(1);
+					options.Resolver.GetFormatterWithVerify<string>().Serialize(ref writer, value.Name, options);
+				}
+			}
+			""";
+
+		string fixedSource = /* lang=c#-test */ """
+			using MessagePack;
+			using MessagePack.Formatters;
+			using Nerdbank.MessagePack;
+			using PolyType;
+			
+			[MessagePackConverter(typeof(MyTypeFormatter))]
+			public class MyType
+			{
+				public string Name { get; set; }
+			}
+
+			[GenerateShape<string>]
+			public partial class MyTypeFormatter : MessagePackConverter<MyType>
+			{
+				public override MyType Deserialize(ref Nerdbank.MessagePack.MessagePackReader reader, SerializationContext context)
+				{
+					if (reader.TryReadNil())
+					{
+						return null;
+					}
+
+					string name = null;
+					context.DepthStep();
+					int count = reader.ReadArrayHeader();
+					for (int i = 0; i < count; i++)
+					{
+						switch (i)
+						{
+							case 0:
+								name = context.GetConverter<string, MyTypeFormatter>().Deserialize(ref reader, context);
+								break;
+							default:
+								reader.Skip(context);
+								break;
+						}
+					}
+
+					return new MyType { Name = name };
+				}
+
+				public override void Serialize(ref Nerdbank.MessagePack.MessagePackWriter writer, in MyType value, SerializationContext context)
+				{
+					if (value is null)
+					{
+						writer.WriteNil();
+						return;
+					}
+
+					writer.WriteArrayHeader(1);
+					context.GetConverter<string, MyTypeFormatter>().Serialize(ref writer, value.Name, context);
+				}
+			}
+			""";
+
+		await this.VerifyCodeFixAsync(source, fixedSource);
+	}
+
+	[Fact]
+	public async Task MessagePackObject_Keys()
+	{
+		string source = /* lang=c#-test */ """
+			using MessagePack;
+			using MessagePack.Formatters;
+
+			[{|NBMsgPack102:MessagePackObject|}]
+			public class MyType
+			{
+				[{|NBMsgPack103:Key(0)|}]
```

Comment:
@sharwell this and the `MessagePackObject_Map` test are failing because it takes 2 iterations instead of 1. 

I expect it to be able to be fixed in just 1 iteration as well, since these are two distinct diagnostic IDs, both emitted initially, and the code fix provider is written to offer code fixes all the diagnostics at once.
But your test harness only delivers one diagnostic at a time to the code fix provider. I don't know if that's why we get 2 iterations. What should I do here?